workspaces/global-header/plugins/global-header/src/components/HeaderDropdownComponent/HeaderDropdownComponent.test.tsx

```diff
@@ -0,0 +1,120 @@
+/*
+ * Copyright Red Hat, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import React from 'react';
+import { screen, fireEvent } from '@testing-library/react';
+import { HeaderDropdownComponent } from './HeaderDropdownComponent';
+import { renderInTestApp } from '@backstage/test-utils';
+
+describe('HeaderDropdownComponent', () => {
+  const mockOnOpen = jest.fn();
+  const mockOnClose = jest.fn();
+
+  beforeEach(() => {
+    jest.clearAllMocks();
+
+    jest.spyOn(console, 'warn').mockImplementation(message => {
+      if (message.includes('⚠️ React Router Future Flag Warning')) {
+        return; // Suppress React Router warning
+      }
+      // eslint-disable-next-line no-console
+      console.warn(message); // Log other warnings
+    });
+
+    jest.spyOn(console, 'error').mockImplementation(message => {
```

Comment:
this is the first time I've seen the suppression of these warnings and errors appear in this repo. 

is it absolutely necessary for tests to pass to suppress them?

Reply:
Not necesary for the tests to pass. Just less noise because of the warnings and errors. i can remove this if needed.

Reply:
I personally really like that code. All the warnings in unit tests (and also in the browser console) are super annoying.

I think we should have this in more tests. The goal should be that you can run tests without any warning!